.devcontainer/Dockerfile

```diff
@@ -1,18 +1,20 @@
-FROM mcr.microsoft.com/vscode/devcontainers/dotnet:1-8.0-jammy
-
-USER vscode
-WORKDIR /home/vscode
+FROM debian:bookworm
 
 # Install necessary packages
 RUN set -eux \
-    && wget -q https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb \
-    && sudo dpkg -i packages-microsoft-prod.deb \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends ca-certificates curl \
+    && curl -o packages-microsoft-prod.deb https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb \
+    && dpkg -i packages-microsoft-prod.deb \
+    && apt-get update \
     && rm packages-microsoft-prod.deb \
-    && sudo apt update \
-    && sudo apt-get install -y libmsquic \
-    && sudo rm -rf /var/lib/apt/lists/* \
-    && sudo apt-get clean \
-    && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
-
-# Set environment variables
-ENV PATH=/home/vscode/.cargo/bin:$PATH
+    && apt-get install -y libmsquic git libicu72 build-essential \
+    && curl https://dot.net/v1/dotnet-install.sh -o dotnet-install.sh -L \
+    && chmod +x dotnet-install.sh \
+    && ./dotnet-install.sh -v latest -c 8.0 \
+    && ./dotnet-install.sh -v latest -q preview -c 9.0 \
```

Comment:
I would make downloading the install script and running it a separate RUN step.

Reply:
Split the RUN command into several steps